### PR TITLE
Initial proposal for the js docs around types

### PR DESCRIPTION
**Pull request by lededje**

### CODE REVIEW

File: _guidelines/javascript.md [hunk 1]
```diff
@@ -14,10 +14,50 @@ Taken in part from the following sources:
 
 ## Table of Contents
 
+  1. [Types](#types)
   1. [Commas](#commas)
 
 ## Commas
 
+## Types
```
Comment (jazlalli) [inline _guidelines/javascript.md]: You've got this `Types` stuff under the `Commas` heading. Move it to below all the `Commas` content, or just move the `Commas` heading down.

File: _guidelines/javascript.md [hunk 2]
```diff
@@ -14,10 +14,50 @@ Taken in part from the following sources:
...
+    const foo = 1;
+    let bar = foo;
+
+    bar = 9;
+
+    console.log(foo, bar); // => 1, 9
+    ```
+
+  <a name="types--complex"></a><a name="1.2"></a>
+  - [Complex](#types--complex): When you access a complex type you work on a reference to its value.
```
Comment (mezis) [inline _guidelines/javascript.md]: Why? A line or two of rationale might help here.

**Comment by mezis** [comment]
@lededje: care to fix conflicts and merge this?

**Comment by lededje** [comment]
With sparrow out I can fix this up and submit some more of them

**Comment by martyphee** [comment]
Looks like you need to merge master in.  Otherwise it looks good.

**Pull request merged**